sonic_platform_base/sonic_xcvr/mem_maps/public/cmis.py

```diff
@@ -22,111 +22,394 @@ def __init__(self, codes):
+            CodeRegField(consts.MODULE_STATE, self.getaddr(0x0, 3), self.codes.MODULE_STATE,
+                 *(RegBitField("Bit%d" % (bit), bit) for bit in range (1, 4))
+            ),
+            NumberRegField(consts.MODULE_FIRMWARE_FAULT_INFO, self.getaddr(0x0, 8), size=1),
+            NumberRegField(consts.MODULE_FLAG_BYTE1, self.getaddr(0x0, 9), size=1),
+            NumberRegField(consts.MODULE_FLAG_BYTE2, self.getaddr(0x0, 10), size=1),
+            NumberRegField(consts.MODULE_FLAG_BYTE3, self.getaddr(0x0, 11), size=1),
+            NumberRegField(consts.CDB1_STATUS, self.getaddr(0x0, 37), size=1),
+            CodeRegField(consts.MODULE_FAULT_CAUSE, self.getaddr(0x0, 41), self.codes.MODULE_FAULT_CAUSE),
+        )
+
+        self.TRANS_PM = RegGroupField(consts.TRANS_PM_FIELD,
+            NumberRegField(consts.VDM_SUPPORTED_PAGE, self.getaddr(0x2f, 128),
+                *(RegBitField("Bit%d" % (bit), bit) for bit in range (0, 2))
+            ),
+            NumberRegField(consts.VDM_CONTROL, self.getaddr(0x2f, 144), size=1, ro=False),
+        )
+
+        self.MEDIA_LANE_FEC_PM = RegGroupField(consts.MEDIA_LANE_FEC_PM,
+            NumberRegField(consts.RX_BITS_PM, self.getaddr(0x34, 128), format=">Q", size=8),
+            NumberRegField(consts.RX_BITS_SUB_INTERVAL_PM, self.getaddr(0x34, 136), format=">Q", size=8),
+            NumberRegField(consts.RX_CORR_BITS_PM, self.getaddr(0x34, 144), format=">Q", size=8),
+            NumberRegField(consts.RX_MIN_CORR_BITS_SUB_INTERVAL_PM, self.getaddr(0x34, 152), format=">Q", size=8),
+            NumberRegField(consts.RX_MAX_CORR_BITS_SUB_INTERVAL_PM, self.getaddr(0x34, 160), format=">Q", size=8),
+            NumberRegField(consts.RX_FRAMES_PM, self.getaddr(0x34, 168), format=">I", size=4),
+            NumberRegField(consts.RX_FRAMES_SUB_INTERVAL_PM, self.getaddr(0x34, 172), format=">I", size=4),
+            NumberRegField(consts.RX_FRAMES_UNCORR_ERR_PM, self.getaddr(0x34, 176), format=">I", size=4),
+            NumberRegField(consts.RX_MIN_FRAMES_UNCORR_ERR_SUB_INTERVAL_PM, self.getaddr(0x34, 180), format=">I", size=4),
+            NumberRegField(consts.RX_MAX_FRAMES_UNCORR_ERR_SUB_INTERVAL_PM, self.getaddr(0x34, 184), format=">I", size=4),
             # TODO: add other PMs...
         )
 
-        self.MEDIA_LANE_LINK_PM = RegGroupField(consts.MEDIA_LANE_LINK_PM_FIELD,
-            NumberRegField(consts.RX_AVG_CD_PM_FIELD, self.get_addr(0x35, 128), format=">i", size=4),
-            NumberRegField(consts.RX_MIN_CD_PM_FIELD, self.get_addr(0x35, 132), format=">i", size=4),
-            NumberRegField(consts.RX_MAX_CD_PM_FIELD, self.get_addr(0x35, 136), format=">i", size=4),
-            NumberRegField(consts.RX_AVG_DGD_PM_FIELD, self.get_addr(0x35, 140), format=">H", size=2, scale=100)
+        # MEDIA_LANE_LINK_PM block corresponds to all the coherent PMs in C-CMIS spec reported in Page 35h.
+        self.MEDIA_LANE_LINK_PM = RegGroupField(consts.MEDIA_LANE_LINK_PM,
```

Comment:
can you move c-cmis specific code to c-cmis mem-map file?.

Reply:
opened new file named c_cmis.py for c-cmis specific registers defined in OIF-C-CMIS-01.1